src/chart_types/wordcloud/renderer/svg/connected_component.tsx

```diff
@@ -125,36 +128,74 @@ function layoutMaker(config: Configs, data: Datum[]) {
     .fontSize((d: Word) => getFontSize(d));
 }
 
-const View = ({ words, conf }: { words: Word[]; conf: Configs }) => (
-  <svg width={getWidth(conf)} height={getHeight(conf)} role="presentation">
-    <g transform={`translate(${getWidth(conf) / 2}, ${getHeight(conf) / 2})`}>
-      {words.map((d, i) => {
-        return (
-          <text
-            key={String(i)}
-            style={{
-              fontSize: getFontSize(d),
-              fontStyle: getFontStyle(d),
-              fontFamily: getFont(d),
-              fontWeight: getFontWeight(d),
-              fill: d.color,
-            }}
-            textAnchor="middle"
-            transform={`translate(${d.x}, ${d.y}) rotate(${d.rotate})`}
-          >
-            {d.text}
-          </text>
-        );
-      })}
-    </g>
-  </svg>
-);
+const View = ({
+  words,
+  conf,
+  actions: { onElementClick, onElementOver, onElementOut },
+  specId,
+}: {
+  words: Word[];
+  conf: Configs;
+  actions: {
+    onElementClick?: SettingsSpec['onElementClick'];
+    onElementOver?: SettingsSpec['onElementOver'];
+    onElementOut?: SettingsSpec['onElementOut'];
+  };
+  specId: string;
+}) => {
+  return (
+    <svg width={getWidth(conf)} height={getHeight(conf)} role="presentation">
+      <g transform={`translate(${getWidth(conf) / 2}, ${getHeight(conf) / 2})`}>
+        {words.map((d, i) => {
+          const { datum } = d;
+          const actions = {
+            ...(onElementClick && {
+              onClick: () => {
+                onElementClick?.([[datum, { specId, key: specId }]]);
+              },
+            }),
+            ...(onElementOver && {
+              onMouseOver: () => {
+                onElementOver?.([[datum, { specId, key: specId }]]);
+              },
+            }),
+            ...(onElementOut && {
+              onMouseOut: () => {
+                onElementOut?.();
+              },
+            }),
```

Comment:
Unneeded optional chaining

```suggestion
              onClick: () => {
                onElementClick([[datum, { specId, key: specId }]]);
              },
            }),
            ...(onElementOver && {
              onMouseOver: () => {
                onElementOver([[datum, { specId, key: specId }]]);
              },
            }),
            ...(onElementOut && {
              onMouseOut: () => {
                onElementOut();
              },
            }),
```

Reply:
🤦

Reply:
fixed here https://github.com/elastic/elastic-charts/pull/1180/commits/1e8fb67e5bb73d1ae81e69165085e94c52044cbe